blog/docs/articles/from-cloudevent-to-apach-kafka-records-part-one.md

```diff
@@ -0,0 +1,114 @@
+# From CloudEvents to Apache Kafka Records, Part I
+
+**Authors:**
+* **Daniele Zonca, Senior Principal Software Engineer @ Red Hat**
+* **Matthias Weßendorf, Principal Software Engineer @ Red Hat**
+
+**Date: 2023-03-01**
+
+_In this blog post you will learn how to store incoming CloudEvents to an Apache Kafka Topic using the KafkaSink component._
+
+
+Apache Kafka is used in a lot of very different use cases but the need to adopt Kafka protocol can be a barrier especially when there are third party components with limited extension possibilities.
+
+There are producer of events that doesn't support Kafka protocol and HTTP can be a more flexible option. Strimzi project has a [Bridge component](https://strimzi.io/docs/bridge/latest/) that exposes producer/consumer API via HTTP but it is specific for Kafka so it is essentially the same protocol (with consumer group, offset, etc).
+
+Do you think CloudEvents requirement might be an issue? CloudEvents defines a [binding also for HTTP format](https://github.com/cloudevents/spec/blob/v1.0.2/cloudevents/bindings/http-protocol-binding.md#31-binary-content-mode) also and especially in binary mode, most of HTTP payload are possibly already a valid CloudEvents!
+
+The [Knative Sink for Apache Kafka](https://knative.dev/docs/eventing/sinks/kafka-sink/) is a Kafka-native implementation for a CloudEvent ingress and persisting the event as a Apache Kafka Record on a configurable topic.
+
+### Setting up the Apache Kafka Topic
+
+In order use the `KafkaSink` component you need to have a topic for Apache Kafka with the proper access to it. For this post we are using a local Apache Kafka installation, powered by [Strimzi](https://strimzi.io), as described [here](https://knative.dev/blog/articles/single-node-kafka-development/). Once the Apache Kafka cluster is running in your Kubernetes environment it is time to create the topic. For this, we are using the `KafkaTopic` CRD from Strimzi to create a topic in a standard declarative Kubernetes way:
+
+```yaml
+apiVersion: kafka.strimzi.io/v1beta2
+kind: KafkaTopic
+metadata:
+  name: my-topic
+  namespace: kafka
+  labels:
+    strimzi.io/cluster: my-cluster
+spec:
+  partitions: 1
+  replicas: 1
+```
+
+This will create a simple topic, with `partitions` and `replicas` both set to `1`, which is **not** recommended on a production environment. Once the manifest has been applied to the Kubernetes cluster it can be queried like:
```

Comment:
would a side note or link on what are the recommendation for production environment be a good addition so that I, as a reader, learn best practices?

Reply:
we could just rephrase the text and take a setup like: https://knative.dev/blog/articles/single-node-kafka-development/ 

as a given (which is for development)

Reply:
Updated and added link to a article that discusses the prod config for the kafka based broker impl.